crates/astria-core/src/protocol/abci.rs

```diff
@@ -1,76 +1,49 @@
-use std::{
-    borrow::Cow,
-    num::NonZeroU32,
-};
+use std::num::NonZeroU32;
+
+use tendermint::abci::Code;
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 #[allow(clippy::module_name_repetitions)]
-pub struct AbciErrorCode(u32);
+pub struct AbciErrorCode(Code);
 
 #[rustfmt::skip]
 impl AbciErrorCode {
-    pub const UNSPECIFIED: Self = Self(0);
-    pub const UNKNOWN_PATH: Self = Self(1);
-    pub const INVALID_PARAMETER: Self = Self(2);
-    pub const INTERNAL_ERROR: Self = Self(3);
-    pub const INVALID_NONCE: Self = Self(4);
-    pub const TRANSACTION_TOO_LARGE: Self = Self(5);
-    pub const INSUFFICIENT_FUNDS: Self = Self(6);
-    pub const INVALID_CHAIN_ID: Self = Self(7);
-    pub const VALUE_NOT_FOUND: Self = Self(8);
-    pub const TRANSACTION_EXPIRED: Self = Self(9);
-    pub const TRANSACTION_FAILED: Self = Self(10);
-    pub const BAD_REQUEST: Self = Self(11);
+    pub const UNKNOWN_PATH: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(1) }));
+    pub const INVALID_PARAMETER: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(2) }));
+    pub const INTERNAL_ERROR: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(3) }));
+    pub const INVALID_NONCE: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(4) }));
+    pub const TRANSACTION_TOO_LARGE: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(5) }));
+    pub const INSUFFICIENT_FUNDS: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(6) }));
+    pub const INVALID_CHAIN_ID: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(7) }));
+    pub const VALUE_NOT_FOUND: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(8) }));
+    pub const TRANSACTION_EXPIRED: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(9) }));
+    pub const TRANSACTION_FAILED: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(10) }));
+    pub const BAD_REQUEST: Self = Self(Code::Err(unsafe { NonZeroU32::new_unchecked(11) }));
 }
 
 impl AbciErrorCode {
+    /// Returns the wrapped [`Code`].
     #[must_use]
-    pub fn info(self) -> Cow<'static, str> {
```

Comment:
I would leave this.

Reply:
Done in [5367a02](https://github.com/astriaorg/astria/pull/1280/commits/5367a0230be034ed20550471579909ba6746b604).